src/realm/object-store/shared_realm.cpp

```diff
@@ -1080,6 +1080,11 @@ bool Realm::compact()
 void Realm::convert(const Config& config, bool merge_into_existing)
 {
     verify_thread();
+
+    if (m_config.sync_config && m_config.sync_config->flx_sync_requested) {
```

Comment:
This check is backwards. This allows attempting to convert to flexible sync and forbids converting from flexible sync, which is the opposite of what we need.

Reply:
The issue says either from or to… so probably this check has to be made more strict.

Reply:
@jbreams can advise on what it is allowed and what it is not? The issue says that we should throw for any type of conversion when flexible sync is involved..

Reply:
I think we can allow converting from a FLX sync realm to a local realm, but not the other way around. Can we make this the reverse of what it was as @tgoyne originally suggested. so `if (config.sync_config && config.sync_config->flx_sync_requested)`

Reply:
done.